bot/cogs/reddit.py

```diff
@@ -42,16 +43,19 @@ def __init__(self, bot: Bot):
         if params is None:
             params = {}
 
-        response = await self.bot.http_session.get(
-            url=f"{self.URL}/{route}.json",
-            headers=self.HEADERS,
-            params=params
-        )
-
-        content = await response.json()
-        posts = content["data"]["children"]
+        for _ in range(self.MAX_FETCH_RETRIES):
+            response = await self.bot.http_session.get(
+                url=f"{self.URL}/{route}.json",
+                headers=self.HEADERS,
+                params=params
+            )
+            if response.status == 200 and response.content_type == 'application/json':
+                # Got appropriate response - process and return.
+                content = await response.json()
+                posts = content["data"]["children"]
+                return posts[:amount]
 
-        return posts[:amount]
+        return list()  # Failed to get appropriate response within allowed number of retries.
```

Comment:
Add some debug logging before it returns that indicates an invalid response was received, including the URL, status code, and mimetype.

Reply:
Okay, understood. Just to be clear, you only want to log the details of the final attempt, correct? Rather than logging each failure in the full retry loop? 

Also, would you want any asyncio.sleep() delay in between retries? I'm not sure if that'd make a valid response significantly more likely, but it seems plausible that it might.

Reply:
Yes, just log after the final attempt. A sleep would be good (maybe 3 seconds?) but in that case you should make the bot show as typing when the commands are used so users know they're meant to wait.

Reply:
Okay, I'll get a sleep in there!

Reply:
I've added logging, 3 second sleep, and typing indication during sleep - I did have to modify a few function signatures to achieve the typing indication, just a heads up. I ran the code and tested the commands (including with an intentionally bad route) and it seems to work as intended.